source/docs/software/dashboards/glass/mech2d-widget.rst

```diff
@@ -0,0 +1,108 @@
+The Mechanism2d Widget
+======================
+
+Glass supports displaying stick-figure representations of your robot's mechanisms using the :guilabel:`Mechanism2d` widget. It supports combinations of ligaments that can rotate and / or extend or retract, such as arms and elevators and they can be combined for more complicated mechanisms. An instance of the ``Mechanism2d`` class should be created and populated, sent over NetworkTables, and updated periodically with the latest mechanism states in your robot code. It can also be used with the :doc:`Physics Simulation </docs/software/wpilib-tools/robot-simulation/physics-sim>` to visualize and program your robot's mechanisms before the robot is built.
+
+Creating and Configuring the Mechanism2d Instance
+-------------------------------------------------
+
+The ``Mechanism2d`` object is the "canvas" where the mechanism is drawn. The root node is where the mechanism is anchored to ``Mechanism2d``. For a single jointed arm this would the pivot point. For an elevator, this would be where it's attached to the robot's base. To get a root node (represented by a ``MechanismRoot2d`` object), call ``getRoot(name, x, y)`` on the container ``Mechanism2d`` object. The coordinate system follows the same orientation as Field2d - ``(0,0)`` is bottom left.
+
+In the examples below, an elevator is drawn, with a rotational wrist on top of the elevator. The full mechanism2d example is available in `Java <https://github.com/wpilibsuite/allwpilib/blob/main/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java>`__ / `C++ <https://github.com/wpilibsuite/allwpilib/blob/main/wpilibcExamples/src/main/cpp/examples/Mechanism2d/cpp/Robot.cpp>`__
+
+.. tabs::
+
+  .. group-tab:: Java
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/a446c25598546813ff410654bea343b2d7bc0e6c/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java
+        :language: java
+        :lines: 43-46
+        :linenos:
+        :lineno-start: 43
+
+  .. group-tab:: C++
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/a446c25598546813ff410654bea343b2d7bc0e6c/wpilibcExamples/src/main/cpp/examples/Mechanism2d/cpp/Robot.cpp
+        :language: cpp
+        :lines: 59-62
+        :linenos:
+        :lineno-start: 59
+
+
+Each ``MechanismLigament2d`` object represents a stage of the mechanism. Call ``append()``/``Append()`` on a root node or ligament node to add another node to the figure. In Java, pass a constructed ``MechanismLigament2d`` object to add it. In C++, pass the construction parameters in order to construct and add a ligament.
+
+.. tabs::
+
+  .. group-tab:: Java
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/a446c25598546813ff410654bea343b2d7bc0e6c/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java
```

Comment:
What is `kElevatorMinimumLength`? What units? I see `90` being passed as an argument, what about mechanisms with no angles?

Reply:
added description of parameters